.github/workflows/release_pr.yml

```diff
@@ -14,11 +14,48 @@ jobs:
       - name: Unshallow
         run: git fetch --prune --unshallow
       
-      # git-cliff to generate CHANGELOG.md
+      # Setup
       - uses: actions/setup-python@v5
         with:
           python-version: '3.10'
       - run: pip install git-cliff typos
+      - name: Set git config
+        run: git config user.email "dcn-ecosystem@cisco.com" && git config user.name "dcn-ecosystem"
+
+      # Check new changlog entry for version bump type
+      - name: Find version bump type
+        id: type
+        run: |
+          changelog=$(git cliff --unreleased)
+          case $changelog in
+              *"BREAKING CHANGES:"* )
+                  echo "MAJOR"
+                  echo "bump="MAJOR"" >> $GITHUB_OUTPUT
+                  ;;
+              *"IMPROVEMENTS:"* )
+                  echo "MINOR"
+                  echo "bump="MINOR"" >> $GITHUB_OUTPUT
+                  ;;
+              *"DEPRECATIONS:"* )
+                  echo "MINOR"
+                  echo "bump="MINOR"" >> $GITHUB_OUTPUT
+                  ;;
+              * )
+                  echo "PATCH"
+                  echo "bump="PATCH"" >> $GITHUB_OUTPUT
+                  ;;
+          esac
+
+      # Commit major or minor version change messages
+      # NOTE: Without these commit messages git-cliff will only increase patch version.
+      - name: Commit major version change message
+        if: steps.type.outputs.bump == 'MAJOR'
+        run: 'git commit -m "major: [ignore] Major version change" --allow-empty'
+      - name: Commit minor version change message
+        if: steps.type.outputs.bump == 'MINOR'
+        run: 'git commit -m "minor: [ignore] Minor version change" --allow-empty'

```

Comment:
Is this the only way?

Reply:
I found a way to dynamically insert a commit message into git-cliff without it being in the commit history using the --with-commit argument.
Looks like the most suitable way for the tool to register a major/minor version bump without an empty commit.